mesonbuild/interpreter/interpreter.py

```diff
@@ -3279,6 +3279,18 @@ def build_target_decorator_caller(self, node, args, kwargs):
                     outputs.update(o)
 
         kwargs['include_directories'] = self.extract_incdirs(kwargs)
+
+        if isinstance(targetclass, build.Executable):
+            if kwargs['gui_app'] is not None:
+                if kwargs['win_subsystem'] is not None:
+                    raise InvalidArguments.from_node(
+                        'Executable got both "gui_app", and "win_subsystem" arguments, which are mutually exclusive',
```

Comment:
It would be sweet if this kind of logic could be hoisted into typed_kwargs in the future.